tools/sdk/include/user_interface.h

```diff
@@ -519,9 +529,9 @@ typedef struct {
 } Event_StaMode_AuthMode_Change_t;
 
 typedef struct {
-    struct ip_addr ip;
-    struct ip_addr mask;
-    struct ip_addr gw;
+    struct ipv4_addr ip;
```

Comment:
What will happen here? How will the event work with ipv6?

Reply:
Espressif FW cannot be IPv6-aware (until they reveal their sources). Its internals are IPv4. There is a name collision between espressif's lwIP-v1.4 "struct ip_addr" and lwIP-v2 + IPv6's "struct ip_addr".
I had to rename espressif's structure to make sure legacy ipv4 is used with FW internals when using any core configuration.

Reply:
I'm aware of that, but what will be the arg to these events with IPv6? an invalid IPv4? Can we work around it?

Reply:
WiFi events need testing, and it's possible they won't work with IPv6. But let's figure that out later.